src/state/transactions/reducer.ts

```diff
@@ -66,13 +218,32 @@ export default createReducer(initialState, (builder) =>
       } else {
         tx.lastCheckedBlockNumber = Math.max(blockNumber, tx.lastCheckedBlockNumber)
       }
-    })
-    .addCase(finalizeTransaction, (transactions, { payload: { hash, chainId, receipt } }) => {
+    },
+    finalizeTransaction(transactions, { payload: { hash, chainId, receipt } }) {
       const tx = transactions[chainId]?.[hash]
       if (!tx) {
         return
       }
       tx.receipt = receipt
       tx.confirmedTime = now()
+    },
+  },
+  extraReducers: (builder) => {
+    builder.addCase(updateVersion, (transactions) => {
+      // in case there are any transactions in the store with the old format, remove them
+      Object.keys(transactions).forEach((chainId) => {
+        const chainTransactions = transactions[chainId as unknown as number]
+        Object.keys(chainTransactions).forEach((hash) => {
+          if (!('info' in chainTransactions[hash])) {
+            // clear old transactions that don't have the right format
+            delete chainTransactions[hash]
+          }
+        })
+      })
     })
-)
+  },
+})
+
+export const { addTransaction, clearAllTransactions, checkedTransaction, finalizeTransaction } =
+  transactionSlice.actions
+export default transactionSlice.reducer
```

Comment:
i'm surprised we're doing a default export here but seems like we already were?

Reply:
It was what was there before so I kept it to prevent changes in files that import this. Should I remove the default?

Reply:
nbd